### ISSUE TITLE
Cannot set ngpu=1 to recog_wav.sh

### ISSUE THREAD

**Issue by robothubtokyo** [espnet/espnet#1121]
We could run "run.sh --ngpu 1" and "recog_wav.sh --ngpu 0", but cannot run "recog_wav.sh --ngpu 1". How can we do?
Here is the log.
Thank you.

```
cat decode/gu_csj/log/decode.log
\# asr_recog.py --config conf/decode.yaml --ngpu 1 --backend pytorch --debugmode 1 --verbose 1 --recog-json decode/gu_csj/dump/data.json --result-label decode/gu_csj/result.json --model exp/train_nodup_sp_pytorch_train/results/model.acc.best --rnnlm exp/train_rnnlm_pytorch_lm/rnnlm.model.best 
\# Started at Thu Aug 22 15:21:25 JST 2019
\#
2019-08-22 15:21:25,711 (asr_recog:126) WARNING: CUDA_VISIBLE_DEVICES is not set.
2019-08-22 15:21:25,711 (asr_recog:137) INFO: python path = (None)
2019-08-22 15:21:25,711 (asr_recog:142) INFO: set random seed = 1
2019-08-22 15:21:25,711 (asr_recog:150) INFO: backend = pytorch
/home/robothub/hdd_mount/espnet0.5.0/espnet/tools/venv/lib/python3.7/site-packages/sklearn/externals/joblib/externals/cloudpickle/cloudpickle.py:47: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
2019-08-22 15:21:26,395 (deterministic_utils:24) INFO: torch type check is disabled
2019-08-22 15:21:26,395 (asr_utils:361) INFO: reading a config file from exp/train_nodup_sp_pytorch_train/results/model.json
2019-08-22 15:21:26,396 (asr:266) INFO: reading model parameters from exp/train_nodup_sp_pytorch_train/results/model.acc.best
/home/robothub/hdd_mount/espnet0.5.0/espnet/tools/venv/lib/python3.7/site-packages/torch/nn/_reduction.py:49: UserWarning: size_average and reduce args will be deprecated, please use reduction='none' instead.
  warnings.warn(warning.format(ret))
2019-08-22 15:21:26,718 (asr_utils:361) INFO: reading a config file from exp/train_rnnlm_pytorch_lm/model.json
2019-08-22 15:21:26,840 (asr:595) INFO: gpu id: [0]
2019-08-22 15:21:35,421 (asr:614) INFO: (1/1) decoding gu_csj
Traceback (most recent call last):
  File "/home/robothub/hdd_mount/espnet0.5.0/espnet/egs/csj/asr1/../../../espnet/bin/asr_recog.py", line 169, in <module>
    main(sys.argv[1:])
  File "/home/robothub/hdd_mount/espnet0.5.0/espnet/egs/csj/asr1/../../../espnet/bin/asr_recog.py", line 157, in main
    recog(args)
  File "/home/robothub/hdd_mount/espnet0.5.0/espnet/espnet/asr/pytorch_backend/asr.py", line 647, in recog
    nbest_hyps = model.recognize(feat, args, train_args.char_list, rnnlm)
  File "/home/robothub/hdd_mount/espnet0.5.0/espnet/espnet/nets/pytorch_backend/e2e_asr_transformer.py", line 225, in recognize
    enc_output, _ = self.encoder(feat, None)
  File "/home/robothub/hdd_mount/espnet0.5.0/espnet/tools/venv/lib/python3.7/site-packages/torch/nn/modules/module.py", line 489, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/robothub/hdd_mount/espnet0.5.0/espnet/espnet/nets/pytorch_backend/transformer/encoder.py", line 109, in forward
    xs, masks = self.embed(xs, masks)
  File "/home/robothub/hdd_mount/espnet0.5.0/espnet/tools/venv/lib/python3.7/site-packages/torch/nn/modules/module.py", line 489, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/robothub/hdd_mount/espnet0.5.0/espnet/espnet/nets/pytorch_backend/transformer/subsampling.py", line 36, in forward
    x = self.conv(x)
  File "/home/robothub/hdd_mount/espnet0.5.0/espnet/tools/venv/lib/python3.7/site-packages/torch/nn/modules/module.py", line 489, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/robothub/hdd_mount/espnet0.5.0/espnet/tools/venv/lib/python3.7/site-packages/torch/nn/modules/container.py", line 92, in forward
    input = module(input)
  File "/home/robothub/hdd_mount/espnet0.5.0/espnet/tools/venv/lib/python3.7/site-packages/torch/nn/modules/module.py", line 489, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/robothub/hdd_mount/espnet0.5.0/espnet/tools/venv/lib/python3.7/site-packages/torch/nn/modules/conv.py", line 320, in forward
    self.padding, self.dilation, self.groups)
RuntimeError: Input type (torch.FloatTensor) and weight type (torch.cuda.FloatTensor) should be the same
\# Accounting: time=10 threads=1
\# Ended (code 1) at Thu Aug 22 15:21:35 JST 2019, elapsed time 10 seconds
```

**Comment by sw005320** [comment]
We don't support the GPU decoding for the transformer model for now. If you want to run it with a GPU, please use the RNN model.
@ShigekiKarita, can you add some error handling for this?

**Comment by ShigekiKarita** [comment]
Actually you can use GPU non-batch decoding with arbitrary pytorch models
with "--api v2" option.

2019年8月22日(木) 21:07 Shinji Watanabe <notifications@github.com>:

**Comment by sw005320** [comment]
Oh yes! Then, it's further better if you have error handlings to use `--api v2` in the above case.

**Comment by robothubtokyo** [comment]
Thank you.
We added --api v2 at line.216 in recognizer_wav.sh, and got another error.
Here is the log.

`
\# asr_recog.py --config conf/decode.yaml --ngpu 1 --backend pytorch --debugmode 1 --verbose 1 --recog-json decode/2019-08-22T154318.292Z/dump/data.json --result-label decode/2019-08-22T154318.292Z/result.json --model exp/train_nodup_sp_pytorch_train/results/model.acc.best --api v2 --rnnlm exp/train_rnnlm_pytorch_lm/rnnlm.model.best 
\# Started at Thu Aug 22 15:43:20 UTC 2019
\#
2019-08-22 15:43:20,300 (asr_recog:138) WARNING: CUDA_VISIBLE_DEVICES is not set.
2019-08-22 15:43:20,300 (asr_recog:149) INFO: python path = (None)
2019-08-22 15:43:20,301 (asr_recog:154) INFO: set random seed = 1
2019-08-22 15:43:20,301 (asr_recog:162) INFO: backend = pytorch
/espnet/tools/venv/lib/python3.7/site-packages/sklearn/externals/joblib/externals/cloudpickle/cloudpickle.py:47: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
2019-08-22 15:43:21,271 (recog:30) WARNING: experimental API for custom LMs is selected by --api v2
2019-08-22 15:43:21,271 (deterministic_utils:24) INFO: torch type check is disabled
2019-08-22 15:43:21,271 (asr_utils:327) INFO: reading a config file from exp/train_nodup_sp_pytorch_train/results/model.json
2019-08-22 15:43:21,272 (asr:266) INFO: reading model parameters from exp/train_nodup_sp_pytorch_train/results/model.acc.best
/espnet/tools/venv/lib/python3.7/site-packages/torch/nn/_reduction.py:49: UserWarning: size_average and reduce args will be deprecated, please use reduction='none' instead.
  warnings.warn(warning.format(ret))
2019-08-22 15:43:21,585 (font_manager:1341) INFO: generated new fontManager
Traceback (most recent call last):
  File "/espnet/egs/csj/asr1/../../../espnet/bin/asr_recog.py", line 188, in <module>
    main(sys.argv[1:])
  File "/espnet/egs/csj/asr1/../../../espnet/bin/asr_recog.py", line 171, in main
    recog_v2(args)
  File "/espnet/espnet/asr/pytorch_backend/recog.py", line 39, in recog_v2
    model, train_args = load_trained_model(args.model)
  File "/espnet/espnet/asr/pytorch_backend/asr.py", line 274, in load_trained_model
    torch_load(model_path, model)
  File "/espnet/espnet/asr/asr_utils.py", line 405, in torch_load
    model.load_state_dict(model_state_dict)
  File "/espnet/tools/venv/lib/python3.7/site-packages/torch/nn/modules/module.py", line 769, in load_state_dict
    self.__class__.__name__, "\n\t".join(error_msgs)))
RuntimeError: Error(s) in loading state_dict for E2E:
	Unexpected key(s) in state_dict: "encoder.embed.out.1.pe", "decoder.embed.1.pe". 
\# Accounting: time=2 threads=1
\# Ended (code 1) at Thu Aug 22 15:43:22 UTC 2019, elapsed time 2 seconds
 `

**Comment by kan-bayashi** [comment]
Maybe fixed in #1127
Could you try the latest master?

**Comment by robothubtokyo** [comment]
Thank you.
We tried again and got another error.
Here is the log.
`\# asr_recog.py --api v2 --config conf/decode.yaml --ngpu 1 --backend pytorch --debugmode 1 --verbose 1 --recog-json decode/2019-08-24T020209.684Z/dump/data.json --result-label decode/2019-08-24T020209.684Z/result.json --model exp/train_nodup_sp_pytorch_train/results/model.acc.best --rnnlm exp/train_rnnlm_pytorch_lm/rnnlm.model.best 
\# Started at Sat Aug 24 02:02:11 UTC 2019
\#
2019-08-24 02:02:11,941 (asr_recog:141) WARNING: CUDA_VISIBLE_DEVICES is not set.
2019-08-24 02:02:11,941 (asr_recog:152) INFO: python path = (None)
2019-08-24 02:02:11,941 (asr_recog:157) INFO: set random seed = 1
2019-08-24 02:02:11,941 (asr_recog:165) INFO: backend = pytorch
/espnet/tools/venv/lib/python3.7/site-packages/sklearn/externals/joblib/externals/cloudpickle/cloudpickle.py:47: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
  import imp
2019-08-24 02:02:13,076 (recog:30) WARNING: experimental API for custom LMs is selected by --api v2
2019-08-24 02:02:13,077 (deterministic_utils:24) INFO: torch type check is disabled
2019-08-24 02:02:13,077 (asr_utils:328) INFO: reading a config file from exp/train_nodup_sp_pytorch_train/results/model.json
2019-08-24 02:02:13,077 (asr:292) INFO: reading model parameters from exp/train_nodup_sp_pytorch_train/results/model.acc.best
/espnet/tools/venv/lib/python3.7/site-packages/torch/nn/_reduction.py:49: UserWarning: size_average and reduce args will be deprecated, please use reduction='none' instead.
  warnings.warn(warning.format(ret))
2019-08-24 02:02:13,395 (font_manager:1341) INFO: generated new fontManager
2019-08-24 02:02:13,869 (asr_utils:328) INFO: reading a config file from exp/train_rnnlm_pytorch_lm/model.json
Traceback (most recent call last):
  File "/espnet/egs/csj/asr1/../../../espnet/bin/asr_recog.py", line 191, in <module>
    main(sys.argv[1:])
  File "/espnet/egs/csj/asr1/../../../espnet/bin/asr_recog.py", line 174, in main
    recog_v2(args)
  File "/espnet/espnet/asr/pytorch_backend/recog.py", line 51, in recog_v2
    lm_class = dynamic_import_lm(lm_args.model_module, lm_args.backend)
AttributeError: 'Namespace' object has no attribute 'model_module'
\# Accounting: time=3 threads=1
\# Ended (code 1) at Sat Aug 24 02:02:14 UTC 2019, elapsed time 3 seconds
`

**Comment by kan-bayashi** [comment]
I believe fixed in #1133.

**Comment by robothubtokyo** [comment]
We confirmed that it works normally.
Thank you!
